internal/tofu/node_resource_plan.go

```diff
@@ -438,3 +450,15 @@ func (n *nodeExpandPlannableResource) resourceInstanceSubgraph(ctx context.Conte
 	graph, graphDiags := b.Build(ctx, addr.Module)
 	return graph, diags.Append(graphDiags).ErrWithWarnings()
 }
+
+// Close implements closableResource
+func (n *nodeExpandPlannableResource) Close() (diags tfdiags.Diagnostics) {
+	if n.Addr.Resource.Mode != addrs.EphemeralResourceMode {
+		return diags
+	}
+
+	for _, c := range n.closers {
+		diags = diags.Append(c())
```

Comment:
This is way out of scope and more of a refinement, but as each of these calls will be waiting on a chan/timeout, this could take up to 10 seconds * (count or len(for_each)).

Reply:
You're right! I will take a note on this and have this checked later.